### PR TITLE
One time tripmon

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR adds support for monitoring one-time trips, i.e. trips that are taken only once. A trip that has no monitored day is considered a one-time trip.

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java [hunk 1]
```diff
@@ -294,12 +294,10 @@ public MonitoredTrip updateWeekdays(boolean value) {
     }
 
     /**
-     * Returns true if the trip is not active overall or if all days of the week are set to false
+     * Returns true if the trip is not active.
      */
     public boolean isInactive() {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java]: This methods seems a bit redundant now given that it is just inverting isActive.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 2]
```diff
@@ -571,6 +585,10 @@ private long getMinutesUntilTrip() {
      * calculated and updated in the monitored trip's journey state.
      */
     public boolean shouldSkipMonitoredTripCheck() throws Exception {
+        return shouldSkipMonitoredTripCheck(true);
+    }
+
+    public boolean shouldSkipMonitoredTripCheck(boolean persist) throws Exception {
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java:591]: This is probably out of scope, but this method is hard to follow and would benefit from being broken up into smaller methods. There are also a lot of TODOs, which are also probably out of scope.

File: src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/SkipMonitoringTestArgs.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    /** Whether the trip is one-time or recurring */
+    public final boolean isRecurring;
+    /* A helpful message describing the particular test case */
+    public final String message;
+
+    /**
+     * If true, it is expected that the {@link CheckMonitoredTripTest#createSkipTripTestCases()} method should
+     * calculate that the given trip should be skipped.
+     */
+    public final boolean result;
```
Comment (br648) [inline src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/SkipMonitoringTestArgs.java]: Nit. Could be named `expectedResult`.

File: src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/SkipMonitoringTestArgs.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    public final boolean pastState;
```
Comment (br648) [inline src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/SkipMonitoringTestArgs.java:20]: Comment?

File: src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTripBasicTest.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        if (args.isRecurring) {
+            setMonitoredDaysForTest(trip);
+        }
+        if (args.pastState) {
+            trip.journeyState.tripStatus = TripStatus.PAST_TRIP;
+        }
+        assertEquals(
+            args.result,
+            new CheckMonitoredTrip(trip).shouldSkipMonitoredTripCheck(false),
+            args.message
```
Comment (br648) [inline src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTripBasicTest.java:41]: This is probably OS related, but I don't get the message. Instead I get something like: 

`org.opentripplanner.middleware.tripmonitor.jobs.SkipMonitoringTestArgs@279fedbd`

**Review by br648 — APPROVED** [review]
This looks ok to me. Just some minor comments to address, but approving.

File: src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/SkipMonitoringTestArgs.java (hunk 4 above)
Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/SkipMonitoringTestArgs.java:20]: Updated in https://github.com/ibi-group/otp-middleware/pull/224/commits/cba63c6304330bc6a1532a959f80f670999e43e4.

File: src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/SkipMonitoringTestArgs.java (hunk 3 above)
Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/SkipMonitoringTestArgs.java]: Updated in https://github.com/ibi-group/otp-middleware/pull/224/commits/cba63c6304330bc6a1532a959f80f670999e43e4.

File: src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTripBasicTest.java (hunk 5 above)
Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTripBasicTest.java:41]: That is strange, the third argument of this `assert.Equals` should match the string type.

File: src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java]: I was on the fence for that but I removed `isInactive` in https://github.com/ibi-group/otp-middleware/pull/224/commits/294902194f7099cb07a164eb9f0ec141d2c1fd2c.

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java (hunk 2 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java:591]: I absolutely agree that one is a brain method.

**Review by miles-grant-ibigroup — APPROVED** [review]
Agree with Rob that that one class is a bit noodly but all the new code looks reasonable to me
